vere/main.c

```diff
@@ -587,11 +587,22 @@ main(c3_i   argc,
   printf("~\n");
   //  printf("welcome.\n");
   printf("urbit %s\n", URBIT_VERSION);
-  printf("urbit: home is %s\n", u3_Host.dir_c);
+  int mprint_i = 1000;
+  c3_c* abs_c = (c3_c *)malloc(mprint_i);
+
+  // allocates more memory as needed if the path is too large
+  //
+  while ( abs_c != getcwd(abs_c, mprint_i) ) {
+    free(abs_c);
+    mprint_i *= 2;
+    abs_c = (c3_c *)malloc(mprint_i);
+  }
+  printf("boot: home is %s/%s\n", abs_c, u3_Host.dir_c);
+  free(abs_c);
```

Comment:
Printing an absolute path this way could duplicate segments of the path, include multiple forward slashes, etc. I think the right way to do this is to use [`realpath()`](http://man7.org/linux/man-pages/man3/realpath.3.html), with the second argument NULL so it will allocate internally.